### ISSUE TITLE
Test `initial_disc_too_low` doesn't work as intended, triggering a `numpy` `DeprecationWarning`

### ISSUE THREAD

**Issue by joshuacwnewton** [spinalcordtoolbox/spinalcordtoolbox#4405]
While investigating new DeprecationWarnings in https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/4332 that arose due to an update to the `numpy` package, I came across this:

```
testing/cli/test_cli_sct_label_vertebrae.py::test_sct_label_vertebrae_initial_disc_too_low
  /home/runner/work/spinalcordtoolbox/spinalcordtoolbox/spinalcordtoolbox/labels.py:133: DeprecationWarning: NumPy will stop allowing conversion of out-of-bound Python integers to integer arrays.  The conversion of -1 to uint8 will fail in the future.
  For the old behavior, usually:
      np.array(value).astype(dtype)
  will give the desired result (the cast overflows).
    out.data[x, y, z_rpi] = value
```

This deprecation warning is thrown in the `create_labels_along_segmentation` function here:

https://github.com/spinalcordtoolbox/spinalcordtoolbox/blob/0f2f1125ce76aa94e48a6d0709884fff487fcbd6/spinalcordtoolbox/labels.py#L133

tl;dr: We initialize the `out` image using `zeros_like`, which creates a zeroed `uint8` image, since the source image for `zeros_like` is also `uint8`. Then, we try to assign the nonsense label value `-1` to the uint8 image, where it gets cast to `uint8`, becoming `255`.

```python
>>> arr = np.array([0]).astype(np.uint8)
>>> arr[0] = -1
<input>:1: DeprecationWarning: NumPy will stop allowing conversion of out-of-bound Python integers to integer arrays.  The conversion of -1 to uint8 will fail in the future.
For the old behavior, usually:
    np.array(value).astype(dtype)
will give the desired result (the cast overflows).
>>> print(arr)
[255]
```

Then, miraculously, this results in the correct error, but only because the label value is now too _high_ rather than too _low_, and both result in the same error message.
